crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -1256,6 +1271,162 @@ impl IndexerReader {
         Ok(name_bcs_value)
     }
 
+    pub async fn compose_sui_transaction_block_response(
+        &self,
+        stored_transaction: StoredTransaction,
+        options: SuiTransactionBlockResponseOptions,
+        package_resolver: Arc<Resolver<impl PackageStore>>,
+    ) -> IndexerResult<SuiTransactionBlockResponse> {
+        let options = options.clone();
+        let tx_digest = TransactionDigest::try_from(
+            stored_transaction.transaction_digest.as_slice(),
+        )
+        .map_err(|e| {
+            IndexerError::PersistentStorageDataCorruptionError(format!(
+                "Can't convert {:?} as tx_digest. Error: {e}",
+                stored_transaction.transaction_digest
+            ))
+        })?;
+
+        let sender_signed_data_option = if options.show_input || options.show_object_changes {
```

Comment:
mostly moved from `transactions.rs` other than object_changes parts, the move was necessary b/c get_object_changes needs an ObjectProvider 
https://github.com/MystenLabs/sui/blob/5505e624feccbd623f130358653e1fe613fd3181/crates/sui-json-rpc/src/object_changes.rs#L14